CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -591,37 +629,50 @@ public static function getMembershipRevenueAmount($lineItem) {
    * Create transaction for deferred revenue.
    *
    * @param array $lineItems
-   *
-   * @param CRM_Contribute_BAO_Contribution $contributionDetails
-   *
+   * @param int|CRM_Contribute_BAO_Contribution $contribution
    * @param bool $update
-   *
    * @param string $context
    *
+   * @return void
    */
-  public static function createDeferredTrxn($lineItems, $contributionDetails, $update = FALSE, $context = NULL) {
-    if (empty($lineItems)) {
+  public static function createDeferredTrxn($lineItems, $contribution, $update = FALSE, $context = NULL) {
```

Comment:
Please revert the change from $contributionDetails to $contribution. Make the calling context look up the contribution from the contribution id if necessary.